rig/place_and_route/utils.py

```diff
@@ -136,3 +141,18 @@ def build_routing_tables(routes, net_keys, omit_default_routes=True):
             )
 
     return routing_tables
+
+
+class MultisourceRouteError(Exception):
+    """Indicates that two nets with the same key and mask would cause packets
+    to become duplicated.
+    """
+    def __init__(self, key, mask, coordinate):
+        self.key = key
+        self.mask = mask
+        self.x, self.y = coordinate
+
+    def __str__(self):
+        return ("Two different nets with the same key ({0.key:#010x}) and "
+                "mask ({0.mask:#010x}) fork differently at ({0.x}, {0.y})".
```

Comment:
As in alternative redundant routes (hot spares?) could be loaded which
would be enabled by removing the other entries. Maybe?
On 28 Oct 2015 09:37, "Andrew Mundy" <notifications@github.com> wrote:

> In rig/place_and_route/utils.py
> <https://github.com/project-rig/rig/pull/192#discussion_r43233382>:
>
> > @@ -136,3 +141,18 @@ def build_routing_tables(routes, net_keys, omit_default_routes=True):
> >              )
> >
> >      return routing_tables
> > +
> > +
> > +class MultisourceRouteError(Exception):
> > +    """Indicates that two nets with the same key and mask would cause packets
> > +    to become duplicated.
> > +    """
> > +    def __init__(self, key, mask, coordinate):
> > +        self.key = key
> > +        self.mask = mask
> > +        self.x, self.y = coordinate
> > +
> > +    def __str__(self):
> > +        return ("Two different nets with the same key ({0.key:#010x}) and "
> > +                "mask ({0.mask:#010x}) fork differently at ({0.x}, {0.y})".
>
> I'm not sure I understand what you mean by "unmasked", sorry.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/project-rig/rig/pull/192/files#r43233382>.
>